pkg/alertmanager/amcfg.go

```diff
@@ -212,21 +233,33 @@ type ConfigBuilder struct {
 	enforcer  enforcer
 }
 
-func NewConfigBuilder(logger *slog.Logger, amVersion semver.Version, store *assets.StoreBuilder, matcherStrategy monitoringv1.AlertmanagerConfigMatcherStrategy) *ConfigBuilder {
+func NewConfigBuilderInNamespace(logger *slog.Logger, amVersion semver.Version, store *assets.StoreBuilder, amNamespace string, matcherStrategy monitoringv1.AlertmanagerConfigMatcherStrategy) *ConfigBuilder {
```

Comment:
do we need a new function name? it might also be simpler to just pass the Alertmanager object?

Reply:
This is why the project itself is v0.x, we don't promise any stability guarantee for the Go code, only for the CRDs.